### PR TITLE
Add functionality to reports#webinar_details_report

### PR DESCRIPTION
**Pull request by jkloian**
Added functionality for reports#webinar_details_report and a spec test.

### CODE REVIEW

File: lib/zoom/actions/report.rb [hunk 1]
```diff
@@ -40,9 +40,9 @@ def meeting_polls_report(*args)
       end
 
       def webinar_details_report(*args)
-        # TODO: implement webinar_details_report
-        # params = Zoom::Params.new(Utils.extract_options!(args))
-        raise Zoom::NotImplemented, 'webinar_details_report is not yet implemented'
+        params = Zoom::Params.new(Utils.extract_options!(args))
+        params.require(:id).permit(%i[page_size next_page_token])
```
Comment (paderinandrey) [inline lib/zoom/actions/report.rb:44]: page_size & next_page_token are redundant params

Comment (kyleboe) [reply lib/zoom/actions/report.rb:44]: I don't believe they are. But I also can't find any place in the docs where these keys are referenced.

Am I looking in the right place? https://marketplace.zoom.us/docs/api-reference/zoom-api/reports/reportwebinardetails

Comment (paderinandrey) [reply lib/zoom/actions/report.rb:44]: Yes. For this endpoint, the only required parameter is id

Comment (paderinandrey) [reply lib/zoom/actions/report.rb:44]: For example, for endpoint [/report/webinars/{webinarId}/participants](https://marketplace.zoom.us/docs/api-reference/zoom-api/reports/reportwebinarparticipants), these parameters are allowed

Comment (kyleboe) [reply lib/zoom/actions/report.rb:44]: I see. Yep you're totally right. Should just be
```suggestion
        params.require(:webinar_id)
```

File: lib/zoom/actions/report.rb [hunk 2]
```diff
@@ -40,9 +40,9 @@ def meeting_polls_report(*args)
...
+        Utils.parse_response self.class.get("/report/webinars/#{params[:id]}", query: params.except(:id), headers: request_headers)
```
Comment (kyleboe) [inline lib/zoom/actions/report.rb:45]: Update to match id name change
```suggestion
        Utils.parse_response self.class.get("/report/webinars/#{params[:webinar_id]}", headers: request_headers)
```

File: spec/lib/zoom/actions/report/webinar_details_report_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Zoom::Actions::Report do
+  let(:zc) { zoom_client }
+  let(:args) { { id: '123456789' } }
```
Comment (kyleboe) [inline spec/lib/zoom/actions/report/webinar_details_report_spec.rb:7]:
```suggestion
  let(:args) { { webinar_id: '123456789' } }
```

File: spec/lib/zoom/actions/report/webinar_details_report_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  describe '#webinar_details_report' do
+    context 'with a valid response' do
+      before :each do
+        stub_request(
+          :get,
+          zoom_url("/report/webinars/#{args[:id]}")
```
Comment (kyleboe) [inline spec/lib/zoom/actions/report/webinar_details_report_spec.rb:14]:
```suggestion
          zoom_url("/report/webinars/#{args[:webinar_id]}")
```

File: spec/lib/zoom/actions/report/webinar_details_report_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        ).to_return(status: 200,
+                    body: json_response('report', 'webinar'),
+                    headers: { 'Content-Type' => 'application/json' })
+      end
+
+      it "requires a 'id' argument" do
+        expect { zc.webinar_details_report(filter_key(args, :id)) }.to raise_error(Zoom::ParameterMissing, [:id].to_s)
```
Comment (kyleboe) [inline spec/lib/zoom/actions/report/webinar_details_report_spec.rb:21]:
```suggestion
        expect { zc.webinar_details_report(filter_key(args, :webinar_id)) }.to raise_error(Zoom::ParameterMissing, [:webinar_id].to_s)
```

**Review by kyleboe — CHANGES_REQUESTED** [review]
Some naming changes and param permission changes

**Comment by joelmichael** [comment]
@jkloian I know you submitted this PR a year ago, but if you want to review the changes requested, and merge master in to resolve the conflicts, I will then merge this into master so long as the tests pass. We are going to release a new version of zoom_rb soon and this can make it in if so.

**Comment by joelmichael** [comment]
I'm not sure about replacing `id` with `webinar_id` at this time... the `webinar_participants_report` method already exists and uses `id` and in fact many other methods use `id` even where the official API specifies a variable of a different name (e.g. `webinarId` in this case). I do agree it would be best to match the API as precisely as possible but don't want to break things by making `webinar_participants_report` take a different argument than people are used to. Ultimately this PR adds one new endpoint and a test for it (that being `webinar_details_report`) and I think it is good to merge. The new `args_to_params` method could perhaps be used for backwards compatibility in a later refactor to make the args closer to the official API sec, or it could just be required (with a lot of advanced notice) that the args be changed.

I went ahead and permitted `include_fields` in `webinar_participants_report` as well, technically unrelated to this PR but I saw it in the API spec and it should be harmless.

Pinging @kyleboe for input

**Pull request closed without merging**